src/client/options/mod.rs

```diff
@@ -192,6 +192,63 @@ impl fmt::Display for StreamAddress {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[non_exhaustive]
+pub enum ServerApiVersion {
+    Version1,
+}
+
+impl ServerApiVersion {
+    pub fn from_string (version: &str) -> Self {
+        match version {
+            "1" => Self::Version1,
+            _ => panic!("Invalid server API version given")
+        }
+    }
+
+    pub(crate) fn as_str(&self) -> &str {
+        match self {
+            Self::Version1 => "1",
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, TypedBuilder)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+pub struct ServerApi {
+    /// The version string of the declared API version
+    pub version: ServerApiVersion,
+
+    /// Whether the server should return errors for features that are not part of the API version
+    pub strict: Option<bool>,
+
+    /// Whether the server should return errors for deprecated features
+    pub deprecation_errors: Option<bool>,
+}
+
+impl ServerApi {
+    pub(crate) fn append_to_command(&self, command: &mut Document) {
```

Comment:
Technically we need this kind of API:
```rust
    pub(crate) fn append_to_command(&self, command: &mut Command) {
        self.append_to_command_body(command.name.as_str(), &mut command.body);
    }

    pub(crate) fn append_to_command_body(&self, command_name: &str, body: &mut Document) {
        if !self.applies_to_command(command_name) {
            return;
        }

        body.insert("apiVersion", self.version.as_str());

        if let Some(strict) = self.strict {
            body.insert("apiStrict", strict);
        }

        if let Some(deprecation_errors) = self.deprecation_errors {
            body.insert("apiDeprecationErrors", deprecation_errors);
        }
    }
```

`append_to_command()` would be used in `Message::with_command` where we have a `Command` instance, while `append_to_command_body` would be used in `Executor::execute_operation_on_connection`, where we operate on a `Document` instance. The advantage would be that the calling code doesn't need to know that they have to check `applies_to_command` first (as is currently the case).

NB: `apply_to_command` may be a more feasible name, but I feel like it's too early for bikeshedding.

Reply:
I think the cleanest way to do this would likely be to add a method to `Command` similar to what we have for [sessions, cluster time, and transaction numbers](https://github.com/mongodb/mongo-rust-driver/blob/4eb171cd412a4d4227bd3c6690838a3a84b5158c/src/cmap/conn/command.rs#L49-L62). The general idea behind our `Command` type is to be a thin wrapper around `Document` that avoids the need to hardcode various field names that we have to access in multiple places. This seems like a good fit for setting the API version of a command.

Reply:
Rather than having separate `Command::set_server_api` and this `ServerApi::append_to_command`, could we consolidate on a single `Command::set_server_api`?

Reply:
This was previously extracted because of where and how API version was injected into commands. I can remove the `impl` for `ServerApi` and move the logic to `Command` if that's more appropriate.